app/helpers/language_helper.py

```diff
@@ -13,17 +14,18 @@
 }
 
 
-def handle_language() -> None:
+def handle_language(metadata: Optional[Mapping[str, Any]] = None) -> None:
+    metadata = metadata or get_metadata(current_user)
     session_store = get_session_store()
+    schema_name = metadata.get("schema_name") if metadata else None
+
     if session_store and (session_data := session_store.session_data):
-        launch_language = session_data.launch_language_code or DEFAULT_LANGUAGE_CODE
+        launch_language = session_data.language_code or DEFAULT_LANGUAGE_CODE
         # pylint: disable=assigning-non-slot
-        g.allowed_languages = get_allowed_languages(
-            session_data.schema_name, launch_language
-        )
+        g.allowed_languages = get_allowed_languages(schema_name, launch_language)
         request_language = request.args.get("language_code")
         if request_language and request_language in g.allowed_languages:
-            session_store.session_data.language_code = request_language
+            session_data.language_code = request_language
```

Comment:
We should keep this how it was `session_store.session_data.language_code`. If session data returned a copy, this would fail. It is usually not a good idea to updates vars that are technically referencing the same memory. It is clearer to update what we want at the source.